packages/vite/src/node/utils.ts

```diff
@@ -534,6 +534,38 @@ export function removeDirSync(dir: string) {
   }
 }
 
+const REMOVE_DIR_TIMEOUT = 5000
+const rmdir = promisify(fs.rm ?? fs.rmdir) // TODO: Remove after support for Node 12 is dropped
+export async function removeDir(dir: string) {
```

Comment:
I think we should rename this to `gracefulRemove` and move it close to `gracefulRename`? We would also want to run this on Windows only, similar to the `renameDir` variable.

Reply:
I have renamed and added the windows branch. Unfortunately this has resulted in an additional `TODO: Remove after support for Node 12 is dropped` :(